api/src/data/migrations/017.create-hazards.ts

```diff
@@ -0,0 +1,28 @@
+import * as knex from "knex";
+
+export async function up(knex: knex.Knex) {
+    await knex.schema.createTable("hazards", function(table) {
+        table.increments("id").primary().notNullable();
+        table.integer("hazard_type_id").notNullable();
+        table.string("description", 4096).nullable();
+        table.string("location", 8).nullable();
+        table.string("location_detail", 8).nullable();
+        table.string("department", 8).nullable();
+        table.string("scope", 8).nullable();
+        table.string("sensitivity", 8).nullable();
+        table.datetime("created_at").nullable().defaultTo(knex.fn.now());
+        table.string("status", 8).nullable();
+        table.integer("reopen_count").notNullable();
+
+        table.foreign("hazard_type_id").references("hazard_types.id");
+        table.foreign("location").references("locations.code");
+        table.foreign("department").references("departments.code");
+        table.foreign("scope").references("scopes.code");
+        table.foreign("sensitivity").references("sensitivities.code");
+        table.foreign("status").references("hazard_statuses.code");
+    });
+};
```

Comment:
See comment above about standard foreign key reference naming.
These should all be
`xxx_code` or preferably `xxx_id` once the primary id column exists in the foreign table.

Reply:
I think per Peter's design, we will just use code as the primary key and not use an `id` surrogate key. I feel that since they are used a lookup tables, the code designated as primary key is sufficient.

Reply:
@datajohnson I will concede that the "id" isn't really neccessary, but I still argue that they should match the reference.
e.g.
```
table.foreign("sensitivity_code").references("sensitivities.code");
```

Otherwise its easy to lose track of whats a real value, and what is a reference to a real value.